### PR TITLE
Composition, style related improvements

### PR DESCRIPTION
**Pull request by FilipLeitner**
## Pull request type

Please check the type of change your PR introduces:

- [x] Bugfix
- [ ] Feature
- [ ] Dependency updates
- [ ] Code style update (formatting, renaming)
- [x] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe)
- [ ] I am unsure (we'll look into it together)

## Do you introduce a breaking change?

- [ ] Yes
- [x] No
- [ ] I am unsure (no worries, we'll find out)

## Checklist

- [x] I understand and agree that the changes in this PR will be licensed under the [MIT License]
- [ ] I have followed the [guidelines for contributing](https://github.com/hslayers/hslayers-ng/blob/master/CONTRIBUTING.md)
- [ ] The proposed change fits to the content of the [code of conduct](https://github.com/hslayers/hslayers-ng/blob/master/CODE_OF_CONDUCT.md)
- [ ] I have added or updated tests and documentation, and the test suite passes (run `npm test` locally)
- [ ] I'm lost; why do I have to check so many boxes? Please help!

### CODE REVIEW

**Review by jmacura — CHANGES_REQUESTED** [review]
Thanks for the many fixes! I am willing to merge, but without the change in the hslayers-server proxy as it is now. See my comment.

File: projects/hslayers-server/src/proxy.js [hunk 1]
```diff
@@ -15,62 +15,60 @@ const cors_proxy = require('cors-anywhere').createServer({
...
-        req.url = decodeURIComponent(req.url);
-        const [base, queryParams] = req.url.split('?');
-        const params = querystring.parse(queryParams);
-        req.url = base + '?' + querystring.encode(params);
-        if (req.url.indexOf('api.geonames.org/searchJSON') > -1) {
+      } else {
+        // req.url = decodeURIComponent(req.url);
+        // const [base, queryParams] = req.url.split('?');
+        // const params = querystring.parse(queryParams);
+        // req.url = base + '?' + querystring.encode(params);
```
[the remark this answers is not in the document]
Comment (FilipLeitner) [reply projects/hslayers-server/src/proxy.js:35]: Didnt really want to push this. Thanks for pointing it out

File: projects/hslayers/src/components/compositions/layer-parser/layer-parser.service.ts [hunk 2]
```diff
@@ -227,7 +196,27 @@ export class HsCompositionsLayerParserService {
    * @param lyr_def - Layer definition object
    * @returns Ol Image or Tile layer
    */
-  createXYZLayer(lyr_def) {
+  async createXYZLayer(lyr_def) {
+    lyr_def.url = decodeURIComponent(lyr_def.url);
+    /**
+     * FIXME : NOT 100% reliable need to be revised along with
```
Comment (jmacura) [inline projects/hslayers/src/components/compositions/layer-parser/layer-parser.service.ts:202]: Is this FIXME still relevant @FilipLeitner ? If so, could you create a separate issue for it?

File: projects/hslayers/src/components/save-map/save-map.service.ts [hunk 3]
```diff
@@ -427,7 +436,7 @@ export class HsSaveMapService {
       const definition = getDefinition(layer);
       if (definition && definition.url) {
         json.protocol = {
-          url: encodeURIComponent(definition.url),
+          url: `${definition.url}/geoserver/${getWorkspace(layer)}/wfs`,
```
Comment (jmacura) [inline projects/hslayers/src/components/save-map/save-map.service.ts:439]: this change, i.e. commit 0c0a2e18366ba8bf79d12c75a56773611e964179 can go away from this PR as it is properly fixed by commit 494a21e94cf66e1a8cf7effea9c9b0c63e5e5e00 in #4330

Comment (FilipLeitner) [reply projects/hslayers/src/components/save-map/save-map.service.ts]: Dropped

**Review by jmacura — APPROVED** [review]

**Pull request merged**